pageserver/src/tenant/mgr.rs

```diff
@@ -262,14 +268,31 @@ pub async fn shutdown_all_tenants() {
     for (tenant_id, tenant) in tenants_to_shut_down {
         join_set.spawn(
             async move {
+                let (_guard, shutdown_progress) = completion::channel();
                 let freeze_and_flush = true;
+                if let Err(progress) = tenant.shutdown(shutdown_progress, freeze_and_flush).await {
+                    drop(_guard);
```

Comment:
Why is dropping the `_guard` here necessary? ISTM that `shutdown_progress` is dropped already and we got the previously started barrier in the form of `progress`.

Reply:
> Additionally, now you glance this and be sure that the following await does not deadlock on it's own guard.

But the `progress` is another barrier than `shutdown_progress`.
That's why I found the explicit `drop` so confusing.

Reply:
The `shutdown(_, _)` could had (eyeballing just by signature) given back a clone to what was `shutdown_progress`.

Reply:
I now hid the drop in 9e5766de921db4f103bf0296a9062529e161818b with the block so one does not have to worry about it.

Reply:
Assuming this is fine.